app/controllers/admin/talentlms_controller.rb

```diff
@@ -7,42 +7,47 @@
 module Admin
   class TalentlmsController < ApplicationController
     before_action :require_can_manage_config!
+    before_action :set_config, only: [:update, :edit, :destroy]
 
     def index
-      @config = config_scope.first_or_initialize
-      redirect_to new_admin_talentlm_path if @config.new_record?
-      redirect_to edit_admin_talentlm_path(@config) unless @config.new_record?
+      @configs = config_scope.order(:subdomain, :id)
     end
 
     def new
-      @config = config_scope.first_or_initialize
+      @config = config_scope.new
     end
 
     def create
       @config = config_scope.create(config_params)
-      respond_with(@config, location: admin_talentlms_path(@config))
+      respond_with(@config, location: admin_talentlms_path)
     end
 
     def edit
-      @config = config_scope.first_or_initialize
     end
 
     def update
-      @config = config_scope.first_or_initialize
       @config.update(config_params)
-      respond_with(@config, location: edit_admin_talentlm_path(@config))
+      respond_with(@config, location: admin_talentlms_path)
+    end
+
+    def destroy
+      @config.courses.each(&:destroy)
+      @config.destroy
```

Comment:
This is fine, but could also be accomplished by adding `dependent: :destroy` to the courses relation on config.
https://guides.rubyonrails.org/v3.2.2/association_basics.html

Reply:
Thanks! Updated this in https://github.com/greenriver/hmis-warehouse/pull/4858/commits/2c8e52bb860a7362c6f607a4f49e98bc646bb3c3